packages/js/client-config-builder/src/bundles/default.ts

```diff
@@ -0,0 +1,171 @@
+import { ClientConfigBuilder } from "../ClientConfigBuilder";
+import { BuilderConfig } from "../types";
+import * as ipfsHttpClient from "./embeds/ipfs-http-client/wrap";
+import * as ipfsResolver from "./embeds/async-ipfs-resolver/wrap";
+
+import { IWrapPackage, Uri } from "@polywrap/core-js";
+import { PluginPackage } from "@polywrap/plugin-js";
+import {
+  ethereumProviderPlugin,
+  Connection,
+  Connections,
+} from "@polywrap/ethereum-provider-js";
+import { httpPlugin } from "@polywrap/http-plugin-js";
+import { fileSystemPlugin } from "@polywrap/fs-plugin-js";
+import { loggerPlugin } from "@polywrap/logger-plugin-js";
+import { concurrentPromisePlugin } from "concurrent-plugin-js";
+import { ExtendableUriResolver } from "@polywrap/uri-resolver-extensions-js";
+import { IUriRedirect } from "@polywrap/core-js";
+
+// $start: getDefaultConfig
+export const ipfsProviders: string[] = [
+  "https://ipfs.wrappers.io",
+  "https://ipfs.io",
+];
+
+interface IDefaultEmbed {
+  uri: Uri;
+  package: IWrapPackage;
+  source: Uri;
+}
+
+interface IDefaultEmbeds {
+  ipfsHttpClient: IDefaultEmbed;
+  asyncIpfsResolver: IDefaultEmbed;
+}
+
+export const embeds: IDefaultEmbeds = {
+  ipfsHttpClient: {
+    uri: Uri.from("embed/ipfs-http-client@1.0.0"),
+    package: ipfsHttpClient.wasmPackage,
+    source: Uri.from("ens/wraps.eth:ipfs-http-client@1.0.0"),
+  },
+  asyncIpfsResolver: {
+    uri: Uri.from("embed/async-ipfs-uri-resolver-ext@1.0.0"),
+    package: ipfsResolver.wasmPackage,
+    source: Uri.from("ens/wraps.eth:async-ipfs-uri-resolver-ext@1.0.0"),
+  },
+};
+
+type UriResolverExtBootloader = [IDefaultEmbed, IUriRedirect, ...Uri[]];
+
+export const uriResolverExts: UriResolverExtBootloader = [
+  embeds.asyncIpfsResolver,
+  {
```

Comment:
I think this redirect is a bit out of place. I ended up writing out my thought process weighing the pros and cons of this strategy:

**For ipfs uri and object**
I think there is an argument to be made for using the ipfs uri and avoiding the redirect altogether. This would probably coincide with using an object instead of an array for the resolver uris.

One point is that the redirect makes it a bit harder for users to copy the default config (as a basis for their own custom configuration) because they can't just copy the whole array like `...uriResolverExts`.

An advantage of using an object with named keys, instead of an array, is that it's more clear what you're accessing. A key named `ensTextRecordResolver` can indicate what the ipfs uri represents. And there is no need to worry about values changing order or anything like that. Users can always do `Object.values(...)` if they want an array.

**For ens uri and array**
On the other hand, the ens uri is going to be much nicer for debugging because the uri resolution history will contain the uri and not an object key. Seeing an ipfs hash instead of a named URI could confuse anyone who isn't familiar with the default config. 

Using an array also means we don't have to worry about key names. Refactoring the resolvers, or reordering them, will not generally lead to breaking changes. The exception would be changes to the ipfs resolver or ens text record resolver, since their order in the array is used.

Reply:
Okay this all sounds super valid, but I'm having a bit of a hard time picturing what this would look like. Would you mind creating another PR on top of this work to show what you mean here? 🙏